scripts/top-up-cycles-ledger-account

```diff
@@ -1,38 +1,93 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
+SOURCE_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+
 print_help() {
   cat <<-EOF
-	Adds cyles to a given principal's main account on the local cycles ledger.
 
-	Usage: $(basename "$0") <principal|canister_name> <cycles>
-	E.g.   $(basename "$0") bd3sg-teaaa-aaaaa-qaaba-cai 1000000000
-	Or     $(basename "$0") my_canister 1000000000
+	Adds cyles to a given principal's main account on the cycles ledger.
 
 	Note: The destination account will be topped up by the desired amount minus fees.
+
+	Note: If you have insufficient cycles on mainnet:
+	* Check that you have ICP:
+	  dfx ledger balance --ic
+	* If not, send ICP to this account:
+	  dfx ledger --ic account-id
+	* Convert some ICP to cycles
+	  dfx cycles --ic convert --icp 1
+	* See how many cycles you have now:
+	  dfx cycles --ic balance
 	EOF
 }
 
-[[ "${1:-}" != "--help" ]] || {
+# Source the clap.bash file ---------------------------------------------------
+source "$SOURCE_DIR/clap.bash"
+# Define options
+clap.define short=n long=network desc="dfx network to use" variable=NETWORK default="local"
+clap.define short=c long=cycles desc="Number of cycles" variable=CYCLES default="1T"
+clap.define short=t long=to desc="The destinantion principal or canister name" variable=ACCOUNT default="backend"
+# Source the output file ----------------------------------------------------------
+source "$(clap.build)"
```

Comment:
This is where the argument parser sets NETWORK, CYCLES and ACCOUNT from the cli flags.